### PR TITLE
GA Migration Python PreCommit Sharding Runners Portability Samza

### PR DESCRIPTION
**Pull request by MarcoRob**
As part of the migration of Precommit and Postcommit Jobs from Jenkins to GA in self-hosted runners, this PR contains:

- Migrated workflow [job-precommit-python-runners-portability-samza.yml](https://github.com/apache/beam/blob/f1589cdf35afc78573148b4f58fe7418b0be507c/.github/workflows/job-precommit-python-runners-portability-samza.yml)

The migrated workflow was added to [CI.md](https://github.com/apache/beam/blob/f1589cdf35afc78573148b4f58fe7418b0be507c/CI.md)

`DO NOT MERGE` until the effort to use self-hosted runners is completed https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job-precommit-python-runners-portability-samza.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+name: Python Precommit Runners Portability Samza
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 */6 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: ['v*']
+    paths: ['sdks/python/apache_beam/runners/portability/samza_runner_test.py']
```
Comment (damccorm) [inline .github/workflows/job-precommit-python-runners-portability-samza.yml:29]: Please remove this, postcommits shouldn't be gated on paths

**Review by damccorm — COMMENTED** [review]
Just one comment, otherwise looks good

File: .github/workflows/job-precommit-python-runners-portability-samza.yml (hunk 1 above)
Comment (MarcoRob) [reply .github/workflows/job-precommit-python-runners-portability-samza.yml:29]: Hi @damccorm This is a precommit job, should be applied the same change as well?

Comment (MarcoRob) [reply .github/workflows/job-precommit-python-runners-portability-samza.yml:29]: @damccorm just to validate, should I apply the change considering that this is a pre-commit, not a post-commit?

Comment (MarcoRob) [reply .github/workflows/job-precommit-python-runners-portability-samza.yml:29]: is there any comment on this?

Comment (damccorm) [reply .github/workflows/job-precommit-python-runners-portability-samza.yml:29]: Sorry for the slow response; you're right, this is fine as is.

**Review by damccorm — APPROVED** [review]

**Pull request closed without merging**